joplin/api/schema.py

```diff
@@ -383,120 +478,335 @@ class Meta:
+            return guide_page_global_id
+        except Exception as e:
+            pass
+
+        # TODO: don't catch everything
+        try:
+            official_document_page_global_id = graphene.Node.to_global_id('OfficialDocumentPageNode', self.page_id)
+            return official_document_page_global_id
+        except Exception as e:
+            pass
+
+
+    def resolve_title(self, resolve_info, *args, **kwargs):
         service_page = None
         # TODO: don't catch everything
         try:
             service_page = ServicePage.objects.get(id=self.page_id)
+            return service_page.title
         except Exception as e:
             pass
 
-        return service_page
-
-    def resolve_information_page(self, info):
         information_page = None
         # TODO: don't catch everything
         try:
             information_page = InformationPage.objects.get(id=self.page_id)
+            return information_page.title
         except Exception as e:
             pass
 
-        return information_page
+        guide_page = None
+        # TODO: don't catch everything
+        try:
+            guide_page = GuidePage.objects.get(id=self.page_id)
+            return guide_page.title
+        except Exception as e:
+            pass
+
+        official_document_page = None
+        # TODO: don't catch everything
+        try:
+            official_document_page = OfficialDocumentPage.objects.get(id=self.page_id)
+            return official_document_page.title
+        except Exception as e:
+            pass
+
+    def resolve_slug(self, resolve_info, *args, **kwargs):
+        service_page = None
+        # TODO: don't catch everything
+        try:
+            service_page = ServicePage.objects.get(id=self.page_id)
+            return service_page.slug
+        except Exception as e:
+            pass
+
+        information_page = None
+        # TODO: don't catch everything
+        try:
+            information_page = InformationPage.objects.get(id=self.page_id)
+            return information_page.slug
+        except Exception as e:
+            pass
 
-    def resolve_guide_page(self, info):
         guide_page = None
         # TODO: don't catch everything
         try:
             guide_page = GuidePage.objects.get(id=self.page_id)
+            return guide_page.slug
         except Exception as e:
             pass
 
-        return guide_page
+        official_document_page = None
+        # TODO: don't catch everything
+        try:
+            official_document_page = OfficialDocumentPage.objects.get(id=self.page_id)
+            return official_document_page.slug
+        except Exception as e:
+            pass
+
+    class Meta:
+        model = DepartmentPageRelatedPage
+        interfaces = [graphene.Node]
+
+
+class TopicPageTopPageNode(DjangoObjectType):
+    title = graphene.String()
+    slug = graphene.String()
+    page_id = graphene.ID()
```

Comment:
this works with some data model changes to manage top links for topics on the topic page instead of using "make this a top link" on each page